domain/agentbinary/service/types.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	coreagentbinary "github.com/juju/juju/core/agentbinary"
+)
+
+// Metadata holds the metadata of the agent binary.
+type Metadata struct {
```

Comment:
Based on conversations with @hpidcock and @wallyworld I think we are looking to avoid having one function that can operate over multiple SHA algorithms. I think if we are going to support multiple we will have a default of 384 then transform in another one from 256 to 384.

The reasons for this is it reduces the noise around a structure like this having to check that at least one is not empty and dealing with the fact when multiple are filled out but they might be in conflict.

With the above said I am not sure there is much reason for this structure after the fact?

Reply:
This file is removed.